ml-agents/mlagents/trainers/torch/distributions.py

```diff
@@ -186,11 +186,13 @@ def _create_policy_branches(self, hidden_size: int) -> nn.ModuleList:
             branches.append(branch_output_layer)
         return nn.ModuleList(branches)
 
-    def _mask_branch(self, logits: torch.Tensor, mask: torch.Tensor) -> torch.Tensor:
+    def _mask_branch(
+        self, logits: torch.Tensor, allow_mask: torch.Tensor
+    ) -> torch.Tensor:
         # Zero out masked logits, then subtract a large value. Technique mentionend here:
-        # https://arxiv.org/abs/2006.14171. Our implementation is ONNX and Barrcuda-friendly.
-        flipped_mask = 1.0 - mask
-        adj_logits = logits * mask - 1e8 * flipped_mask
+        # https://arxiv.org/abs/2006.14171. Our implementation is ONNX and Barracuda-friendly.
+        mask = -1.0 * allow_mask + 1.0
```

Comment:
Please add a comment that the `-1 * x + 1` format is to workaround a bug, so that someone doesn't "optimize" it in the future.